client/src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from "react";
+import styles from "../Carousel/Carousel.module.css";
+import "react-responsive-carousel/lib/styles/carousel.min.css";
+import { Carousel } from "react-responsive-carousel";
+import TaskStatusChart from "../TaskStatusChart/TaskStatusChart";
+import PropTypes from "prop-types";
+
+export const InsightsCarousel = ({ taskData }) => {
+  const startedTasks = {
+    early: [],
+    onTime: [],
+    late: [],
+  };
+
+  const EndedTasks = {
+    early: [],
+    onTime: [],
+    late: [],
+  };
+
+  const summaryTasks = {
+    completed: [],
+    onGoing: [],
+    notStarted: [],
+  };
+
+  const manageStartedTasks = (
+    sampleTaskData,
+    startedTasks,
+    EndedTasks,
+    summaryTasks
+  ) => {
+    const totalTasks = sampleTaskData.length;
+
+    for (const t of sampleTaskData) {
+      const actualTime = Math.floor(t.startDate.getTime() / 60000);
+      const plannedTime = Math.floor(t.originalStartDate.getTime() / 60000);
+
+      const actualEndTime = Math.floor(t.endDate.getTime() / 60000);
+      const plannedEndTime = Math.floor(t.originalEndDate.getTime() / 60000);
+
+      if (actualTime === plannedTime) {
+        startedTasks.onTime.push(t);
+      } else if (actualTime < plannedTime - 5) {
```

Comment:
Just wondering, what is the purpose of the -5, +5? (Or is this just to simply represent a task that was started before, after, etc.?

Reply:
The purpose of the -5 and +5 is give a 5 minute leeway for the user to start their task, rather than judging them at exactly the planned time.

Reply:
Noted